daemon/SecretsImpl/secretsdatabase.cpp

```diff
@@ -63,7 +63,7 @@ static const char *createSecretsTable =
         "   UnlockSemantic INTEGER NOT NULL,"
         "   CustomLockTimeoutMs INTEGER NOT NULL,"
         "   AccessControlMode INTEGER NOT NULL,"
-        "   FOREIGN KEY (CollectionName) REFERENCES Collections(CollectionName),"
+        "   FOREIGN KEY (CollectionName) REFERENCES Collections(CollectionName) ON DELETE CASCADE,"
```

Comment:
We cannot cascade to the KeyEntries table which is... unfortunate.  I will have to add a trigger to delete from the KeyEntries table, unless someone can think of a better solution..?

Reply:
I guess one alternative is to have "SecretsKeyEntries" for keys which are stored in a Secrets storage plugin, and "CryptoKeyEntries" for keys which are stored by a Crypto plugin directly.  Then the SecretsKeyEntries CollectionName can be a proper foreign key constraint with cascade...

Reply:
I've now unified the codepaths so that every Key is a Secret in a Collection, so the constraints can be enforced appropriately.